RecoHGCal/TICL/plugins/HGCGraph.h

```diff
@@ -0,0 +1,40 @@
+// Author: Felice Pantaleo - felice.pantaleo@cern.ch
+// Date: 11/2018
+// Copyright CERN
+
+#ifndef __RecoHGCal_TICL_HGCGraph_H__
+#define __RecoHGCal_TICL_HGCGraph_H__
+
+#include <vector>
+#include "PatternRecognitionbyCAConstants.h"
+#include "HGCDoublet.h"
+
+class HGCGraph {
+ public:
+  void makeAndConnectDoublets(const ticl::patternbyCA::tilePatternRecognitionByCA &h,
```

Comment:
Is HGCGraph really specific to ```PatternRecognitionbyCA``` class implementation?
Would the other methods deriving from PatternRecognitionAlgoBase have another ```makeAndConnectDoublets``` with a different signature or is the type behind ```ticl::patternbyCA::tilePatternRecognitionByCA``` supposed to be generic?
If it's generic, it is named incorrectly and should go to a more common place.

Reply:
Ciao @slava77 
at the current state of development, I cannot really say if this will be the only "tool" used by the CA and, also, if there will be other PR other than CA.